### PR TITLE
10x speed up of full run by caching imported packages

### PR DESCRIPTION
**Pull request by mikedanese**
depends on #7.

Only the last commit is under review.

### CODE REVIEW

**Comment by rmmh** [comment]
c

File: gazel.go [hunk 1]
```diff
@@ -88,6 +92,32 @@ func NewVenderor(root, cfgPath string, dryRun bool) (*Venderor, error) {
...
+}
+
+type icacheVal struct {
+	pkg *build.Package
+	err error
+}
+
+func (v *Venderor) importPkg(path string, srcDir string) (*build.Package, error) {
+	k := icacheKey{path: path, srcDir: srcDir}
+	v.icacheGuard.RLock()
```
Comment (ixdy) [inline gazel.go]: what's the need for the guard here? is something secretly using goroutines behind-the-scenes?

**Review by ixdy — COMMENTED** [review]
LG, just one question

File: gazel.go (hunk 1 above)
Comment (mikedanese) [reply gazel.go]: @ixdy no I think this is from when I tested multiple go routines but didn't see any speedup, I'll remove the locking.

**Pull request merged**